ably/realtime_channel_spec_test.go

```diff
@@ -469,6 +469,97 @@ func TestRealtimeChannel_RTL6c5_NoImplicitAttach(t *testing.T) {
 	}
 }
 
+func TestRealtimeChannel_RTL2f_HandleResume(t *testing.T) {
+	t.Parallel()
+	const channelRetryTimeout = 123 * time.Millisecond
+
+	setup := func(t *testing.T) (
+		in, out chan *proto.ProtocolMessage,
+		c *ably.Realtime,
+		channel *ably.RealtimeChannel,
+		stateChanges ably.ChannelStateChanges,
+		afterCalls chan ablytest.AfterCall,
+	) {
+		in = make(chan *proto.ProtocolMessage, 1)
+		out = make(chan *proto.ProtocolMessage, 16)
+		afterCalls = make(chan ablytest.AfterCall, 1)
+		now, after := ablytest.TimeFuncs(afterCalls)
+
+		c, _ = ably.NewRealtime(
+			ably.WithToken("fake:token"),
+			ably.WithAutoConnect(false),
+			ably.WithNow(now),
+			ably.WithAfter(after),
+			ably.WithChannelRetryTimeout(channelRetryTimeout),
+			ably.WithDial(ablytest.MessagePipe(in, out)),
+		)
+
+		in <- &proto.ProtocolMessage{
+			Action:            proto.ActionConnected,
+			ConnectionID:      "connection-id",
+			ConnectionDetails: &proto.ConnectionDetails{},
+		}
+
+		err := ablytest.Wait(ablytest.ConnWaiter(c, c.Connect, ably.ConnectionEventConnected), nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		channel = c.Channels.Get("test")
+
+		ctx, cancel := context.WithCancel(context.Background())
+		go channel.Attach(ctx)
+		defer cancel()
+
+		ablytest.Instantly.Recv(t, nil, out, t.Fatalf) // Consume ATTACH
+
+		stateChanges = make(ably.ChannelStateChanges, 10)
+		channel.OnAll(stateChanges.Receive)
+
+		return
+	}
+
+	flags := make(map[proto.Flag]string)
+	flags[0] = "flag has presence is provided"
+	flags[1] = "flag has_backlog is provided"
+	flags[2] = "flag resumed is provided"
+	flags[4] = "flag transient is provided"
```

Comment:
I don't really see the point of testing all possible flags fields in which the RESUMED flag isn't set.

Reply:
Okay, I will remove some of the flags, and will just keep a few ( to check negative test cases)